java/java.hints/src/org/netbeans/modules/java/hints/jdk/CheckRegex.java

```diff
@@ -0,0 +1,194 @@
+                            @ConstraintVariableType(variable="$pattern", type="java.lang.String")
+                        }),
+        @TriggerPattern(value="java.util.regex.Pattern.compile($pattern, $flags)",
+                        constraints={
+                            @ConstraintVariableType(variable="$pattern", type="java.lang.String"),
+                            @ConstraintVariableType(variable="$flags", type="int")
+                        }),
+        @TriggerPattern(value="java.util.regex.Pattern.matches($pattern, $text)",
+                        constraints={
+                            @ConstraintVariableType(variable="$pattern", type="java.lang.String"),
+                            @ConstraintVariableType(variable="$text", type="java.lang.CharSequence")
+                        }),
+        @TriggerPattern(value="$str.split($pattern)",
+                        constraints={
+                            @ConstraintVariableType(variable="$str", type="java.lang.String"),
+                            @ConstraintVariableType(variable="$pattern", type="java.lang.String")
+                        }),
+        @TriggerPattern(value="$str.split($pattern, $limit)",
+                        constraints={
+                            @ConstraintVariableType(variable="$str", type="java.lang.String"),
+                            @ConstraintVariableType(variable="$pattern", type="java.lang.String"),
+                            @ConstraintVariableType(variable="$limit", type="int")
+                        }),
+        @TriggerPattern(value="$str.matches($pattern)",
+                        constraints={
+                            @ConstraintVariableType(variable="$str", type="java.lang.String"),
+                            @ConstraintVariableType(variable="$pattern", type="java.lang.String")
+                        }),
+        @TriggerPattern(value="$str.replaceFirst($pattern, $repl)",
+                        constraints={
+                            @ConstraintVariableType(variable="$str", type="java.lang.String"),
+                            @ConstraintVariableType(variable="$pattern", type="java.lang.String"),
+                            @ConstraintVariableType(variable="$repl", type="java.lang.String")
+                        }),
+        @TriggerPattern(value="$str.replaceAll($pattern, $repl)",
+                        constraints={
+                            @ConstraintVariableType(variable="$str", type="java.lang.String"),
+                            @ConstraintVariableType(variable="$pattern", type="java.lang.String"),
+                            @ConstraintVariableType(variable="$repl", type="java.lang.String")
+                        })
+    })
+ 
+    @Messages("ERR_CheckRegex=Check regular expression")
+    public static ErrorDescription computeWarning(HintContext ctx) {
+        
+        String originalString = null;
+        Tree leaf = ctx.getVariables().get("$pattern").getLeaf();
```

Comment:
Please add `// NOI18N`.

Reply:
Thanks for your input. Have made the changes.